File: backend/src/main/java/com/pharmacy/backend/service/ProfileService.java
package com.pharmacy.backend.service;

import com.pharmacy.backend.dto.request.ProfileRequest;
import com.pharmacy.backend.dto.response.ApiResponse;
import com.pharmacy.backend.dto.response.ProfileResponse;

import java.util.List;

public interface ProfileService {
    ApiResponse<List<ProfileResponse>> getUserProfiles();
    ApiResponse<ProfileResponse> createProfile(ProfileRequest request);
    ApiResponse<ProfileResponse> updateProfile(Long id, ProfileRequest request);
    ApiResponse<Void> deleteProfile(Long id);
    ApiResponse<ProfileResponse> getProfileById(Long id);
}

File: frontend-user/src/services/profileService.js
import api from './api';
import { ProfileTransform } from '../utils/dataTransform';

export const profileService = {
  getUserProfiles: async () => {
    const response = await api.get('/profiles');
    return response.data;
  },

  getProfileById: async (profileId) => {
    const response = await api.get(`/profiles/${profileId}`);
    return response.data;
  },

  createProfile: async (profileData) => {
    const requestData = ProfileTransform.toBackendRequest(profileData);
    
    const response = await api.post('/profiles', requestData);
    return response.data;
  },

  updateProfile: async (profileId, profileData) => {
    const requestData = ProfileTransform.toBackendRequest(profileData);
    
    const response = await api.put(`/profiles/${profileId}`, requestData);
    return response.data;
  },

  deleteProfile: async (profileId) => {
    const response = await api.delete(`/profiles/${profileId}`);
    return response.data;
  }
};
